### PR TITLE
Implement scale breaks and the capability to skip days off 

### PR DESCRIPTION
**Pull request by roman-simionov**
This PR implements the following features:
- auto scale breaks on a value axis:
```javascript
valueAxis: {
   autoBreaksEnabled: true,
   maxAutoBreakCount: 3
}
```
- custom scale breaks on both axes:
```javascript
valueAxis: { // or argumentAxis
    breaks: [[300, 700], [1020, 1400]]
}
```
- scale breaks' appearance customization:
```javascript
commonAxisSettings: {
    breakStyle: { 
        width: 5,
        color: "#ababab",
        line: "waved" // or "straight"
    }
}
```
- the capability to skip holidays and days off on a DateTime argument axis:
```javascript
argumentAxis: {
   workdaysOnly: true,   
   workWeek: [1, 2, 3, 4, 5],  //default value
   holidays: [new Date(2017, 11, 25)]
}
```

### CODE REVIEW

File: js/viz/axes/base_axis.js [hunk 1]
```diff
@@ -62,6 +67,123 @@ function getTickGenerator(options, incidentOccurred) {
...
+            }
+            visibleRange -= ranges[i].length;
+            breaks.push({ from: ranges[i].start, to: ranges[i].end });
+            minDiff = RANGE_RATIO * visibleRange;
+        } else {
+            break;
+        }
+    }
+
+    breaks.sort(function(a, b) {
```
Comment (alexanderPolosatov) [inline js/viz/axes/base_axis.js]: Can be replaced to sortingBreaks method usage

**Review by alexanderPolosatov — CHANGES_REQUESTED** [review]

File: js/viz/axes/datetime_breaks.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        return !workdays.some(function(workDay) {
+            return workDay === day;
+        });
+    }).map(function(day) {
+        return days.indexOf(day);
+    });
+}
+
+function getNextDayIndex(dayIndex) {
+    return (++dayIndex) % 7;
```
Comment (alexanderPolosatov) [inline js/viz/axes/datetime_breaks.js]: Why you change value of dayIndex? It's equal to `(dayIndex + 1) % 7; ` expression.

File: js/viz/docs/doccharts.js [hunk 3]
```diff
@@ -1623,6 +1659,34 @@ var dxChart = {
...
+        * @default false
+        */
+        workdaysOnly: false,
+        /**
+        * @name dxchartoptions_argumentaxis_workweek
+        * @publicName workweek
+        * @type Array<number>
+        * @default [1, 2, 3, 4, 5]
+        */
+        workweek: [1, 2, 3, 4, 5],
```
Comment (lukyanovas) [inline js/viz/docs/doccharts.js]: I see that in [other parts of devexteme](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/) we use camel case notation for this word.

File: js/viz/docs/doccharts.js [hunk 4]
```diff
@@ -1623,6 +1659,34 @@ var dxChart = {
             text: undefined
         },
         /**
+        * @name dxchartoptions_argumentaxis_workdaysonly
+        * @publicName workdaysOnly
+        * @type boolean
+        * @default false
+        */
+        workdaysOnly: false,
```
Comment (lukyanovas) [inline js/viz/docs/doccharts.js:1667]: Check case for this word too.

Comment (Krijovnick) [reply js/viz/docs/doccharts.js:1667]: This word doesn't appear anywhere in the DevExtreme.

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**